ocp_resources/machine_config_pool.py

```diff
@@ -2,8 +2,60 @@
 
 
 class MachineConfigPool(Resource):
+    """
+    MachineConfigPool object. API reference:
+    https://docs.openshift.com/container-platform/4.12/rest_api/machine_apis/machineconfigpool-machineconfiguration-openshift-io-v1.html
+    """
+
     api_group = Resource.ApiGroup.MACHINECONFIGURATION_OPENSHIFT_IO
 
     class Status(Resource.Status):
         UPDATED = "Updated"
         UPDATING = "Updating"
+
+    def __init__(
+        self,
+        name=None,
+        labels=None,
+        machine_config_selector_matches=None,
+        configuration=None,
+        yaml_file=None,
+        node_selector_labels=None,
+        max_unavailable=None,
+        paused=none,
+    ):
+        super().__init__(name=name, yaml_file=yaml_file)
+        self.resource_labels = labels
+        self.configuration = configuration
+        self.machine_config_selector_matches = machine_config_selector_matches
+        self.node_selector_labels = node_selector_labels
+        self.max_unavailable = max_unavailable
+        self.paused = paused
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            if self.resource_labels:
+                self.res["metadata"]["labels"] = self.resource_labels
+
+            self.res["spec"] = {}
+            self.res["spec"]["configuration"] = {}
+            if self.configuration:
```

Comment:
`configuration` is mandatory ; please raise if not passed

Reply:
Actually, it's not mandatory.
You can apply the MCP manifest without the `configuration` specified, and it will be generated automatically.